### PR TITLE
SDIT-2200 Implement move booking for profile details

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/prisonperson/PrisonPersonMoveBookingService.kt [hunk 1]
```diff
@@ -47,30 +53,63 @@ class PrisonPersonMoveBookingService(
...
-      if (latestBooking != null &&
-        latestBooking.findLastModifiedPhysicalAttributes().lastModifiedDateTime() > latestBooking.startDateTime
-      ) {
-        physicalAttributesSyncService.physicalAttributesChanged(toOffenderNo, latestBooking.bookingId, nomisResponse)
-        telemetry += "syncToOffenderDps_HEIGHT" to "true"
-        telemetry += "syncToOffenderDps_WEIGHT" to "true"
-      }
+      nomisResponse.bookings
+        .find { it.bookingId == bookingId }
+        ?.takeIf { it.latestBooking }
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/prisonperson/PrisonPersonMoveBookingService.kt:83]: Fix height/weight such that:
* only sync updates to NOMIS for the bookign that moved
* and only if it is the latest booking, not interested in historical booking moves

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/prisonperson/profiledetails/ProfileDetailsSyncService.kt [hunk 2]
```diff
@@ -39,18 +37,27 @@ class ProfileDetailsSyncService(
...
+      offenderNo = event.offenderIdDisplay,
+      bookingId = event.bookingId,
+    )
+
+  suspend fun profileDetailsPhysicalAttributesChanged(
+    profileType: String,
+    offenderNo: String,
+    bookingId: Long,
+    nomisProfileDetails: PrisonerProfileDetailsResponse? = null,
+  ) {
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/prisonperson/profiledetails/ProfileDetailsSyncService.kt:52]: Refactored this class so we can call the sync. without an event (i.e. directly from the move booking handler)

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/prisonperson/PrisonPersonMoveBookingIntTest.kt [hunk 3]
```diff
@@ -305,3 +426,124 @@ fun nomisPhysicalAttributesToOffenderRemeasured(
...
+            createDateTime = "${bookingStartTime.minusDays(7)}",
+            createdBy = "A_USER",
+            modifiedDateTime = null,
+            modifiedBy = null,
+            auditModuleName = "NOMIS",
+          ),
+        ),
+      ),
+    ),
+  )
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/prisonperson/PrisonPersonMoveBookingIntTest.kt:455]: We only have happy path tests at the moment - next step is to improve the test data builders so other test scenarios are easier to write and comprehend, then I'll get onto some negative testing.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/prisonperson/PrisonPersonMoveBookingService.kt [hunk 4]
```diff
@@ -47,30 +53,63 @@ class PrisonPersonMoveBookingService(
...
+        ?.also { latestBooking ->
+          if (latestBooking.findLastModifiedPhysicalAttributes().lastModifiedDateTime() > latestBooking.startDateTime) {
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/prisonperson/PrisonPersonMoveBookingService.kt:85]: I'm guessing you have check for items copied by the copy booking NOMIS proc their dates are not greater than the booking start data? E.g no shenanigans where the booking start date is midnight but modified date is greater?

Or is modified date for copied records always null ?

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/prisonperson/PrisonPersonMoveBookingService.kt (hunk 4 above)
Comment (mikehalmamoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/prisonperson/PrisonPersonMoveBookingService.kt:85]: Copy booking copies the old audit fields across and booking start time actually includes the time element (booking end doesn't!) - so this if statement *should* always work.

**Pull request merged**